R/query-webdata.R

```diff
@@ -41,4 +41,33 @@ setMethod(f = "query",signature("webdata",'missing'),
           definition = function(.Object, field, ...){
             stop('specify a field to query against for webdata object')
           }
-)
\ No newline at end of file
+)
+#'@rdname query
+#'@aliases query,webdata-method  
+setMethod(f = "query",signature("character",'missing'),
+          definition = function(.Object, field, ...){
+            field <- match.arg(.Object, c('webdata'))
+            values <- do.call(paste0(field,"_query"), list(...))
+            return(values)
+          }
+)
+
+#' @importFrom httr content POST content_type_xml
+webdata_query <- function(csw_url = 'https://www.sciencebase.gov/catalog/item/54dd2326e4b08de9379b2fb1/csw'){
+  warning('function in development. Currently ignores all catalogued WCS endpoints. Output is likely to change.')
+  request = '<csw:GetRecords xmlns:csw="http://www.opengis.net/cat/csw/2.0.2" service="CSW" version="2.0.2" resultType="results" outputSchema="http://www.isotc211.org/2005/gmd" maxRecords="1000">
```

Comment:
One thing that might be a good idea here would be to pull out this XML into a separate file rather than declare it inline.  This is especially good if you are doing any extra work in here (like adding parameters to the query) but might be a good idea in general.  Also, is this cool without the xml prolog?